tests/Support/Constraint/CssConstraint.php

```diff
@@ -50,20 +49,28 @@ protected static function getCssNeedleRegularExpressionPattern(string $needle):
     {
         $needleMatcher = \preg_replace_callback(
             self::CSS_REGEX_PATTERN,
-            static function (array $matches): string {
-                switch (true) {
-                    case isset($matches[1]) && $matches[1] !== '':
-                        return '\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
-                    case isset($matches[2]) && $matches[2] !== '':
-                        return '\\s*+';
-                    case isset($matches[3]) && $matches[3] !== '':
-                        return \preg_quote($matches[3], '/') . '\\s*+';
-                    default:
-                        return \preg_quote($matches[0], '/');
-                }
-            },
+            [self::class, 'getCssNeedleRegularExpressionReplacement'],
             $needle
         );
         return '/' . $needleMatcher . '/';
     }
+
+    /**
+     * @param string[] $matches array of matches for {@see CSS_REGEX_PATTERN}
+     *
+     * @return string replacement string, which may be `$matches[0]` if no alteration is needed
+     */
+    private static function getCssNeedleRegularExpressionReplacement(array $matches): string
+    {
+        switch (true) {
+            case isset($matches[1]) && $matches[1] !== '':
+                return '\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
+            case isset($matches[2]) && $matches[2] !== '':
+                return '\\s*+';
+            case isset($matches[3]) && $matches[3] !== '':
```

Comment:
We have this `isset(foo) && foo` construct multiple times here (and also some matches are used multiple times). Would it work to
1. move the matches into local, well-named variables
1. also use the [null coalescing operator](https://www.php.net/manual/de/migration70.new-features.php#migration70.new-features.null-coalesce-op) instead: `$foo = ($matches[1] ?? '') !== ''`;

Reply:
Good point about `isset … &&`…